lib/new_relic/agent/instrumentation/aws_sqs/instrumentation.rb

```diff
@@ -0,0 +1,87 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+module NewRelic::Agent::Instrumentation
+  module AwsSqs
+    MESSAGING_LIBRARY = 'SQS'
+
+    def send_message_with_new_relic(*args)
+      segment = nil
+      begin
+        queue_name = get_queue_name(args[0])
+        segment = NewRelic::Agent::Tracer.start_message_broker_segment(
+          action: :produce,
+          library: MESSAGING_LIBRARY,
+          destination_type: :queue,
+          destination_name: queue_name
+        )
+        add_aws_attributes(segment, queue_name, args[0])
+      rescue => e
+        NewRelic::Agent.logger.error('Error starting message broker segment in Aws::SQS::Client#send_message ', e)
+      end
+      NewRelic::Agent::Tracer.capture_segment_error(segment) do
+        yield
+      end
+    ensure
+      segment&.finish
+    end
+
+    def send_message_batch_with_new_relic(*args)
+      segment = nil
+      begin
+        queue_name = get_queue_name(args[0])
+        segment = NewRelic::Agent::Tracer.start_message_broker_segment(
+          action: :produce,
+          library: MESSAGING_LIBRARY,
+          destination_type: :queue,
+          destination_name: queue_name
+        )
+        add_aws_attributes(segment, queue_name, args[0])
+      rescue => e
+        NewRelic::Agent.logger.error('Error starting message broker segment in Aws::SQS::Client#send_message_batch ', e)
+      end
+      NewRelic::Agent::Tracer.capture_segment_error(segment) do
+        yield
+      end
+    ensure
+      segment&.finish
+    end
+
+    def receive_message_with_new_relic(*args)
+      segment = nil
+      begin
+        queue_name = get_queue_name(args[0])
+        segment = NewRelic::Agent::Tracer.start_message_broker_segment(
+          action: :consume,
+          library: MESSAGING_LIBRARY,
+          destination_type: :queue,
+          destination_name: queue_name
+        )
+        add_aws_attributes(segment, queue_name, args[0])
+      rescue => e
+        NewRelic::Agent.logger.error('Error starting message broker segment in Aws::SQS::Client#receive_message ', e)
+      end
+      NewRelic::Agent::Tracer.capture_segment_error(segment) do
+        yield
+      end
+    ensure
+      segment&.finish
+    end
+
+    def add_aws_attributes(segment, queue_name, params)
+      segment&.add_agent_attribute('messaging.system', 'aws_sqs')
+      segment&.add_agent_attribute('cloud.region', config&.region)
+      segment&.add_agent_attribute('cloud.account.id', get_account_id(params))
```

Comment:
Is Ruby smart enough to not even attempt `get_account_id` if `segment` is `nil`? If not, I think you should guard against `segment` being `nil` with an early return.

Reply:
yeah i can add that a guard, but jsyk it won't call the things in those args if the &. isn't going to happen

Reply:
added a guard here and removed the &. https://github.com/newrelic/newrelic-ruby-agent/pull/2679/commits/fc7f03780c7567ab2bb77b19f6ce9ca8e6e619ac